tests/test_pixeldataset.py

```diff
@@ -8,21 +8,24 @@
 import random
 from pathlib import Path
 from tempfile import TemporaryDirectory
+from unittest import mock
 
 import numpy as np
 import pandas as pd
+import pytest
 from anndata import AnnData
 from numpy.testing import assert_array_equal
 from pandas.testing import assert_frame_equal
 from pixelator.config import AntibodyPanel
-from pixelator.graph import write_recovered_components
+from pixelator.graph import Graph, write_recovered_components
 from pixelator.pixeldataset import (
     FileBasedPixelDatasetBackend,
     ObjectBasedPixelDatasetBackend,
     PixelDataset,
     PixelFileCSVFormatSpec,
     PixelFileFormatSpec,
     PixelFileParquetFormatSpec,
+    _enforce_edgelist_types,
```

Comment:
This was a pragmatic decision on my part here. In general we aren't testing private methods. But in this case I thought it would useful since it's a rather crucial part of data transformation. I'll add a comment about this in the test to explain my reasoning.